### DISCUSSION TITLE
Retrieving Corporate Bonds

### DISCUSSION THREAD

**Discussion by AADias** [alvarobartt/investpy#596] [General]
Hello,
I have used the function:
**search_result = investpy.search_quotes(text='volkswagen', products=['bonds'],
countries=['italy'], n_results=1)**
and I got the result
**{"id_": 1158767, "name": "Volkswagen Leasing GmbH VOWGL 1.125 04-Apr-2024", "symbol": "XS1692347526=TX", "country": "italy", "tag": "/rates-bonds/xs1692347526", "pair_type": "bonds", "exchange": "EuroTLX"}**

But from here I am not able to get the latest prices or historical data for this bond.
How can I do it ?

Thank you

**Comment by ymyke** [comment]
You can use the dict that gets returned to create an investpy `SearchObj` and then call the `retrieve_historical_data` method exposed by that object.

As an alternative, you can use my wrapper package tessa (https://github.com/ymyke/tessa) like so:

```python
import tessa
tessa.price_history(
    query='{"id_": 1158767, "name": "Volkswagen Leasing GmbH VOWGL 1.125 04-Apr-2024", "symbol": "XS1692347526=TX", "country": "italy", "tag": "/rates-bonds/xs1692347526", "pair_type": "bonds", "exchange": "EuroTLX"}',
    type_="searchobj",
)
```

**Reply by AADias** [reply]
Thank you, 
I was actually able to work with your package and also use the SearchObj option.
Using your package the result was just the date and closing price. With the SearchObj I was able to get more info (open, high, low, close and Change %)

![image](https://user-images.githubusercontent.com/30188273/185277256-fb3a999e-858f-48af-a894-c73a22d21d77.png)
![image](https://user-images.githubusercontent.com/30188273/185277603-d63935bb-710f-49af-8fdc-1395edb4a162.png)

Your solution is also able to get all the data from the asset ?

Thank you

**Reply by ymyke** [reply]
That is currently not possible. For that you'd have to go via the investpy searchobj.
